src/providers/uploadHelpers/uploadObservation.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import inatjs from "inaturalistjs";
+import Realm from "realm";
+
+import realmConfig from "../../models/index";
+import { getJWTToken } from "../../components/LoginSignUp/AuthenticationService";
+import ObservationPhoto from "../../models/ObservationPhoto";
+import ObservationSound from "../../models/ObservationSound";
+
+const markRecordUploaded = async ( uuid: string, type: string, response: Object ) => {
+  const { id } = response.results[0];
+
+  try {
+    const realm = await Realm.open( realmConfig );
+    const record = realm.objectForPrimaryKey( type, uuid );
+    realm?.write( ( ) => {
+      record.id = id;
+      record._synced_at = new Date( );
+    } );
+  } catch ( e ) {
+    console.log( e, `couldn't mark ${type} uploaded in realm` );
+  }
+};
+
+const uploadToServer = async ( params, options, uuid, type, apiCall ) => {
+  try {
+    const response = await apiCall.create( params, options );
+    await markRecordUploaded( uuid, type, response );
+  } catch ( e ) {
+    console.log( JSON.stringify( e.response ), `couldn't upload ${type}` );
+  }
+};
+
+const createParams = ( response, options, evidence, mapLocalModelForUpload, type, apiCall ) => {
```

Comment:
This doesn't create params, does it? What is it supposed to do?

Reply:
Would it be more accurate to say this is flattening parameters with the FileUpload class? Maybe `flattenParams` is a better name here?

Reply:
Just reading this method, it looks like it loops over an array of API response results and then uploads photos or sounds to the server with `uploadToUserver`, which doesn't seem like creating params or flattening params... but I'm guessing that's *not* what this method does, so back to my question: what does it do?

Reply:
It's looping over either the local observation photos or observation sounds attached to an observation, then using either `mapPhotoForUpload` or `mapSoundForUpload` to return flattened parameters, then uploading the respective photo or sound.

Reply:
Ok, if the ultimate action is to actually upload the evidence, maybe call it `uploadEvidence` and change the method signature to something like

```
uploadEvidence(
  evidence: array,
  type: string,
  apiSchemaMapper: function,
  baseParams: object,
  apiEndpoint: function
)
```

That would make this more readable to me, at least. Flattening params is a precondition for upload but not, ultimately, the reason you would call this function.